main/fastcgi.c

```diff
@@ -1469,6 +1474,7 @@ int fcgi_accept_request(fcgi_request *req)
 			return -1;
 		}
 		if (fcgi_read_request(req)) {
+			req->hook.on_read();
```

Comment:
I'm not familiar with the fastcgi code, just trying to figure out how all this works...  Looking at the current implementation of [`fcgi_accept_request`](https://github.com/php/php-src/blob/53ff54605495fc2265e99981fbef50edbef47879/main/fastcgi.c#L1358), if fd < 0, we are first calling on_accept() then accept(), then on_read(), then poll() or select(), then fcgi_read_request(). If fd >= 0, then only fcgi_read_request() is called.

This patch changes to call on_read() also after fcgi_read_request(). However, doesn't that mean that in the case of fd < 0, we will end up calling on_read() *twice*? Similarly, should on_read be called before fcgi_read_request() or after?